### ISSUE TITLE
Convert `ShouldBeErrorId` to `Should -Throw -ErrorId` 

### ISSUE THREAD

**Issue by SteveL-MSFT** [PowerShell/PowerShell#6244]
- [ ] Update `Start-PSPester` to check for Pester 4.2+ (need 4.2 for exception passthru support)
- [ ] Convert usage of `ShouldBeErrorId` to `Should -Throw -ErrorId`
  - [ ] Update tests that relied on `ShouldBeErrorId` passthru to `Should -Throw -ErrorId -PassThru`
 -  [ ] Remove `ShouldBeErrorId` custom function
- [ ] Update documentation on how to write tests verifying errors conditions

**Comment by iSazonov** [comment]
Should we fix tests to distinguish terminating/non-terminating errors? We never do this so it is huge work.

Also we don't always use `ShouldBeErrorId`. There's a use of the old template with `try-catch`. I hope we fix its too.

**Comment by SteveL-MSFT** [comment]
Yes, we should fix any `try-catch` error checking to use this pattern as well.

We should defer the terminating/non-terminating problem until Pester has support for it.
